src/schema-manager.ts

```diff
@@ -210,6 +215,37 @@ export class PolygonSchema {
     }
   }
 
+  public async getAllSchemaByDID(did: string) {
+    try {
+      const isValidDid = validateDid(did)
+      if (!isValidDid) {
+        throw new Error('invalid did provided')
+      }
+      const didDetails = await this.resolver.resolve(did)
+
+      if (!didDetails?.didDocumentMetadata?.linkedResourceMetadata) {
+        throw new Error(
+          `The DID linked resource metadata for the given DID was not found!`,
+        )
+      }
+      const linkedResourceMetadata =
+        didDetails?.didDocumentMetadata?.linkedResourceMetadata
+      const schemaList: ResourcePayload[] = linkedResourceMetadata.filter(
+        (element: ResourcePayload) => {
+          if (element.resourceType === 'W3C-schema') {
+            return true
+          }
+          return false
+        },
+      )
```

Comment:
```suggestion
      const schemaList: ResourcePayload[] = linkedResourceMetadata.filter(
        (element: ResourcePayload) => (element.resourceType === 'W3C-schema')
      )
```

Reply:
Thanks for the suggestion, I have changed it